From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: clavin <clavin@DESKTOP-QVNUS47.fastel.com>
Date: Thu, 7 Dec 2023 16:11:40 -0500
Subject: Revert "[api] DCHECK-fail when we reenter V8 while terminating"

This reverts commit 3eed17c6a12a0f33639aa5cffb822ebe42931ec8.

This change causes Electron to crash. :(

This patch is meant to be temporary and should be removed after more
investigation and potentially a refactor.

Xref: https://chromium-review.googlesource.com/c/v8/v8/+/5075293
Xref: https://github.com/electron/electron/issues/40741

diff --git a/src/api/api-macros.h b/src/api/api-macros.h
index a300e2fd324e41e01787eef9a98dd57d52248ce2..8ab1520a20d9c0389ed3dc571974781c41eff31d 100644
--- a/src/api/api-macros.h
+++ b/src/api/api-macros.h
@@ -41,7 +41,6 @@
 
 #define ENTER_V8_HELPER_INTERNAL(i_isolate, context, class_name,               \
                                  function_name, HandleScopeClass, do_callback) \
-  DCHECK(!i_isolate->is_execution_terminating());                              \
   HandleScopeClass handle_scope(i_isolate);                                    \
   CallDepthScope<do_callback> call_depth_scope(i_isolate, context);            \
   API_RCS_SCOPE(i_isolate, class_name, function_name);                         \
@@ -50,7 +49,6 @@
 
 #define PREPARE_FOR_DEBUG_INTERFACE_EXECUTION_WITH_ISOLATE(i_isolate, context, \
                                                            T)                  \
-  DCHECK(!i_isolate->is_execution_terminating());                              \
   InternalEscapableScope handle_scope(i_isolate);                              \
   CallDepthScope<false> call_depth_scope(i_isolate, context);                  \
   i::VMState<v8::OTHER> __state__((i_isolate));                                \
